packages/inject/src/inject.ts

```diff
@@ -0,0 +1,213 @@
+
+  if (!encryptionKey) {
+    encryptionKey = _generateRandomString(32)
+    localStorage.setItem(keyStorage, encryptionKey)
+  }
+
+  const options = encodeURIComponent(JSON.stringify({ config: { ...config, encryptionKey } }))
+  const title = encodeURIComponent(config.botConversationDescription || config.botName || 'Chatbot')
+  const iframeSrc = host + '/index.html?options=' + options
+  const iframeId = _getIframeId(config.chatId)
+  return `<iframe id="${iframeId}" title="${title}" frameborder="0" src="${iframeSrc}" class="bp-widget-web"/>`
+}
+
+function _generateRandomString(length: number) {
+  const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+
+  let str = ''
+  for (let i = 0; i < length; i++) {
+    str += chars.charAt(Math.floor(Math.random() * chars.length))
+  }
+
+  return str
+}
+
+function _makeChatRefProxy(chatId: string, target: Partial<WebchatRef>): WebchatRef {
+  const chatRefHandler: ProxyHandler<WebchatRef> = {
+    get(target, prop: keyof WebchatRef) {
+      if (target[prop]) {
+        return target[prop]
+      }
+
+      if (prop === 'postMessage') {
+        return () => {
+          console.warn(
+            `No webchat with id ${chatId} has been initialized. \n Please use window.botpressWebChat.init first.`
+          )
+        }
+      } else if (prop === 'eventListener') {
+        return {
+          handler: (evnt: WebchatEvent) => console.log('this is the default event handler', evnt),
+          types: []
+        }
+      }
+    },
+    set(target, prop: keyof WebchatRef, value) {
+      target[prop] = value
+      return true
+    }
+  }
+  return new Proxy<Partial<WebchatRef>>(target, chatRefHandler) as WebchatRef
+}
+
+function _getChatRef(chatId?: string) {
+  chatId = chatId || DEFAULT_CHAT_ID
+  return CHAT_REFS[chatId]
+}
+
+function _getIframeElement(containerId: string, iframeId: string): HTMLIFrameElement {
+  return document.querySelector(`#${containerId} #${iframeId}`) as HTMLIFrameElement
+}
+
+function sendEvent(payload: any, chatId?: string) {
+  const chatRef = _getChatRef(chatId)
+  chatRef.postMessage({ action: 'event', payload }, '*')
+}
+
+function sendPayload(payload: any, chatId?: string) {
+  const chatRef = _getChatRef(chatId)
+  chatRef.postMessage({ action: 'sendPayload', payload }, '*')
+}
+
+function configure(payload: Config, chatId?: string) {
+  const chatRef = _getChatRef(chatId)
+  chatRef.postMessage({ action: 'configure', payload }, '*')
+}
+
+function mergeConfig(payload: Partial<Config>, chatId?: string) {
+  const chatRef = _getChatRef(chatId)
+  chatRef.postMessage({ action: 'mergeConfig', payload }, '*')
+}
+
+function onEvent(handler: WebchatEventHandler, topics: WebchatEventHandlerTopics = [], chatId?: string) {
+  if (typeof handler !== 'function') {
+    console.error('eventHandler is not a function, please provide a function')
+    return
+  }
+  if (!topics || typeof topics !== 'object' || !topics.length) {
```

Comment:
```suggestion
  if (!Array.isArray(topics)) {
```

Reply:
good catch this is a leftover of vanilla js